File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.c
int minSubArrayLen(int target, int* nums, int numsSize) {
    int left = 0;
    int curSum = 0;
    int minLen = INT_MAX;

    for (int right = 0; right < numsSize; right++){
        curSum += nums[right];
        while(curSum>=target){
            if(minLen < right-left+1){
                minLen = minLen;
            }
            else{
                minLen = right-left+1;
            }
            curSum-=nums[left];
            left++;
        }
    }
    return minLen == INT_MAX ? 0 : minLen;
}

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.kt
class Solution {
    fun minSubArrayLen(target: Int, nums: IntArray): Int {
        var left = 0
        var curSum = 0
        var minLen = Int.MAX_VALUE

        for(right in nums.indices){
            curSum += nums[right];
            while (curSum >= target){
                minLen = minOf(minLen, right-left+1)
                curSum -= nums[left]
                left++
            }
        }
        return if (minLen == Int.MAX_VALUE) 0 else minLen
    }
}